src/marketplace/types/gmLogsIdl.ts

```diff
@@ -0,0 +1,371 @@
+        },
+        {
+          'name': 'timestamp',
+          'type': 'i64',
+          'index': false
+        }
+      ]
+    },
+    {
+      'name': 'OpenOrdersMemo',
+      'fields': [
+        {
+          'name': 'initializerPubkey',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'openOrdersCount',
+          'type': 'u64',
+          'index': false
+        },
+        {
+          'name': 'depositMint',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'timestamp',
+          'type': 'i64',
+          'index': false
+        }
+      ]
+    },
+    {
+      'name': 'ExchangeMemo',
+      'fields': [
+        {
+          'name': 'orderTaker',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'orderInitializer',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'price',
+          'type': 'u64',
+          'index': false
+        },
+        {
+          'name': 'currencyMint',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'assetMint',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'purchaseQuantity',
+          'type': 'u64',
+          'index': false
+        },
+        {
+          'name': 'orderRemainingQuantity',
+          'type': 'u64',
+          'index': false
+        },
+        {
+          'name': 'timestamp',
+          'type': 'i64',
+          'index': false
+        }
+      ]
+    },
+    {
+      'name': 'CancelOrderMemo',
+      'fields': [
+        {
+          'name': 'initializerPubkey',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'timestamp',
+          'type': 'i64',
+          'index': false
+        }
+      ]
+    }
+  ],
+  'metadata': {
+  }
+}
+
+export const baseIdl: GmLogsIdl =
+{
+  'version': '0.1.0',
+  'name': 'gm_logs',
+  'instructions': [],
+  'events': [
+    {
+      'name': 'RegisterCurrencyMemo',
+      'fields': [
+        {
+          'name': 'updateAuthorityAccount',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'registeredCurrency',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'currencyMint',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'saCurrencyVault',
+          'type': 'publicKey',
+          'index': false
+        },
+        {
+          'name': 'royalty',
+          'type': 'u64',
+          'index': false
+        },
+        {
+          'name': 'timestamp',
+          'type': 'i64',
+          'index': false
+        }
+      ]
+    },
+    {
+      'name': 'InitializeMemo',
+      'fields': [
+        {
+          'name': 'initializerPubkey',
```

Comment:
Seems like a few places where the naming in the IDL here doesn't match the program code. Is that intentional?

```suggestion
          'name': 'orderInitializerPubkey',
```